File: src/main/java/com/jss/jssbackend/modules/miscellaneous/service/BillingItemService.java
package com.jss.jssbackend.modules.miscellaneous.service;

import java.util.List;

import com.jss.jssbackend.modules.miscellaneous.model.BillingItem;

public interface BillingItemService {
    public List<BillingItem> getBillingItems();

    public BillingItem getBillingItem(Integer id);
}

File: client/src/app/modules/miscellaneous/services/billing.item.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/appRest.service';
import { BillingItem } from '../../miscellaneous/model/BillingItem';

@Injectable({
  providedIn: 'root'
})
export class BillingItemService extends AppRestService<BillingItem>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getBillingItems() {
    return this.getList(new HttpParams(), "billing-items");
  }
  
   addOrUpdateBillingItem(billingItem: BillingItem) {
    return this.addOrUpdate(new HttpParams(), "billing-item", billingItem, "Enregistré", "Erreur lors de l'enregistrement");
  }

}
